src/command/toggleattributecommand.js

```diff
@@ -93,16 +96,16 @@ export default class ToggleAttributeCommand extends Command {
 					selection.removeAttribute( this.attributeKey );
 				}
 			} else {
-				const ranges = getSchemaValidRanges( this.attributeKey, selection.getRanges(), document.schema );
+				const workRanges = getSchemaValidRanges( this.attributeKey, selection.getRanges(), document.schema );
 
 				// Keep it as one undo step.
-				const batch = document.batch();
+				const workBatch = batch || document.batch();
```

Comment:
Let's keep the simple naming `batch`. Instead of setting the `batch` in line 85, set it once, here, but to a correct value:

```js
const batch = options.batch || document.batch();
```